RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -808,6 +848,10 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
    for ( const auto& chamber : info.chambers )
    {
      if (chamber.id.subdetId() == 3 && rpcHitHandle_.isValid()  ) continue; // Skip RPC chambers, they are taken care of below)
+
+     if (doME0_){
+       if  (chamber.id.subdetId() == MuonSubdetId::ME0 && me0HitHandle_.isValid()  ) continue; // Skip ME0 chambers, they are taken care of below
```

Comment:
shouldn't this skip happen regardless of doME0_ value?
Otherwise, if doME0_ is false, you will include ME0 chambers here